pkg/utils/utils.go

```diff
@@ -60,6 +63,25 @@ func GetPfAddr(pciAddr string) (string, error) {
 // GetPfName returns SRIOV PF name for the given VF
 // If device is not VF then it will return its own ifname if exist else empty string
 func GetPfName(pciAddr string) (string, error) {
+	pfEswitchMode, err := GetPfEswitchMode(pciAddr)
+	if err != nil {
+		// If device doesn't support eswitch mode query, fall back to the default implementation
+		if strings.Contains(strings.ToLower(fmt.Sprint(err)), "no such device") {
```

Comment:
There is an additional use-case which slipped my mind.

if `GetPfName` is called with a PF (i.e non-sriov) address and it does not have sriov VFs then devlink will fail with 

```
[root ~]# devlink dev eswitch show pci/0000:03:00.0
devlink answers: Operation not supported
```

in this case, we should also use sysfs

Reply:
Added a condition handling this case